### PR TITLE
Eliminate qemu hardcoding and openssl-dev build

### PR DESCRIPTION
**Pull request by ericvh**
Just use rust-optee-trustzone-sdk toolchain and
default openssl-dev packages for aarch64.  These
changes require my enhancement/reorg pull-request as well.

Signed-off-by: Eric Van Hensbergen <eric.vanhensbergen@arm.com>

### CODE REVIEW

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 19f004b47e4e2a6bdbda25226a81fe392b02dee6
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=a5be765c-c97b-46bc-a2aa-1aaac1fac131%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 1ee765393085ab3c92c5e04e7c9d9832c3080e4b
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=916e94e5-51ef-4e54-8727-ae91e9728a96%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: Makefile [hunk 1]
```diff
@@ -29,7 +29,7 @@ sdk:
 	$(MAKE) -C sdk
 
 # Generate all test policy
-test_cases: sdk
+test_cases:
```
Comment (dreemkiller) [inline Makefile:32]: When is this dependency firing? We have trustzone-test-collateral below. Does there ever need to be a test-collateral build without TEE set?

File: runtime-manager/Makefile [hunk 2]
```diff
@@ -128,6 +129,9 @@ TZ_TARGET := aarch64-unknown-optee-trustzone
 
 trustzone: $(FINAL_DIR)/$(TZ_UUID).ta
 
+css-tz.bin: $(FINAL_DIR)/$(TZ_UUID).ta
```
Comment (dreemkiller) [inline runtime-manager/Makefile:132]: when is css-tz.bin used?

**Comment by ericvh** [comment]
should be used as a target now instead of just css. in

**Comment by ericvh** [comment]
sdk was a phony so it always fired

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 271f03c175880b355748303af4e56bc7eae76ace
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=267d0439-0dfc-4339-a188-5fba874da018%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by ericvh** [comment]
tests will continue to fail until the docker changes are also merged.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: c8a8e44b50e5e1aa21bf577f2f522b7870ead171
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=2222cd30-efe1-42b6-a661-af2021d06cbe%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by dreemkiller** [comment]
@ericvh I'm using your veracruz branch, and when executing `make trustzone-veracruz-server-test` I get the following error:

> running: "/work/rust-optee-trustzone-sdk/optee/toolchains/aarch64/bin/aarch64-linux-gnu-gcc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-Wall" "-Wextra" "-o" "/work/veracruz/veracruz-server-test/target/debug/build/libloading-a28a69f7354d0672/out/src/os/unix/global_static.o" "-c" "src/os/unix/global_static.c"
cargo:warning=aarch64-linux-gnu-gcc: error: unrecognized command line option '-m64'
exit code: 1

**Comment by ericvh** [comment]
> @ericvh I'm using your veracruz branch, and when executing `make trustzone-veracruz-server-test` I get the following error:
> 
> > running: "/work/rust-optee-trustzone-sdk/optee/toolchains/aarch64/bin/aarch64-linux-gnu-gcc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-Wall" "-Wextra" "-o" "/work/veracruz/veracruz-server-test/target/debug/build/libloading-a28a69f7354d0672/out/src/os/unix/global_static.o" "-c" "src/os/unix/global_static.c"
> > cargo:warning=aarch64-linux-gnu-gcc: error: unrecognized command line option '-m64'
> > exit code: 1

this sounds like its back to trying to build x86 code on Arm?  At least I think that's where I saw -m64.  I'll try and reproduce, its possible I fixed something by hand that I missed moving over.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: f96544e01bb64d6ec39475394814035b07b8b2ac
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=1132e0ad-8708-457c-b6fc-c2d0e31c7cb1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 8b775efbd8155ac01ad29943929d1904062318da
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=76e8b37a-fde7-4e69-a6f6-df623387f144%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: c5058031b8622339d56ae1d0d5f1d3a7d4119779
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=3236873b-0931-4f6d-aa95-7a5addf01d6a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by dreemkiller — CHANGES_REQUESTED** [review]
There are some generated files (some of which are binary) that have been committed.
Examples include: 
- `veracruz-test/proxy-attestation-server.db`
- `trustzone-root-enclave/trustzone-root-enclave`
- `trustzone-root-enclave/stripped_ta`
- `test-collateral/pgen`
- `sdk/data-generators/number-stream-generator/number-stream-1.txt`
-` sdk/data-generators/number-stream-generator/number-stream-2.txt`
- `sdk/data-generators/linear-regression-generator/linear-regression.txt`
- `runtime-manager/stripped_ta`
- `runtime-manager/runtime_manager_enclave`
They should be removed (and preferably the binary files purged via rebase or something) before we merge.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 268482f28db53a822106ae6ac8ae2e392c380d2c
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=d31939b2-22f7-493f-b133-8f79f4c922b4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by ericvh** [comment]
rebased, currently rerunning clean tests to make sure I didn't mess up anything on the merge.

**Comment by ericvh** [comment]
tests passed, should be ready for merge.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: b68dad41403940c9dfd39b65917a125d2e5bbaa7
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=d0fd35c6-83e4-4f64-b940-16677bcc2a0f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by dreemkiller** [comment]
@ShaleXIONG This is still failing CI, but @ericvh says that it will until the CI updates to the latest docker image changes. Do we want to update the CI to that before this merges, or are we comfortable merging this before updating CI?

**Comment by ShaleXIONG** [comment]
> @ShaleXIONG This is still failing CI, but @ericvh says that it will until the CI updates to the latest docker image changes. Do we want to update the CI to that before this merges, or are we comfortable merging this before updating CI?

I am checking. Will update the Docker image in the CI.

**Comment by ShaleXIONG** [comment]
@dreemkiller  I made minor changes in this pull request. The docker image has been updated on the CI side. It should pass (hopefully). Separately, there is a 'change requested' but not sure if it has been changed.

**Comment by dreemkiller** [comment]
LGTM. Can you run the tests locally to make sure your changes don't break local runs?

**Comment by ShaleXIONG** [comment]
> LGTM. Can you run the tests locally to make sure your changes don't break local runs?

It should not break. 
SGX: with the current version of the SGX image, plus the minor fix I committed in this merge request, it should be all good.
TZ: I have updated the docker a bit (here)[https://github.com/veracruz-project/veracruz-docker-image/pull/18], (add a missing copy of libssl and libcryto), with this change on the docker, TZ is good in my Linux machine and in CI.

**Review by dreemkiller — APPROVED** [review]

**Review by dreemkiller — APPROVED** [review]

**Pull request merged**